src/backend/distributed/metadata/node_metadata.c

```diff
@@ -787,7 +787,7 @@ SyncDistributedObjectsToNode(WorkerNode *workerNode)
 		return;
 	}
 
-	EnsureSequentialModeMetadataOperations();
+	EnsureNoModificationsHaveBeenDone();
```

Comment:
I think we can still keep  `EnsureSequentialModeMetadataOperations` as it also switches to sequential mode. So, maybe :
```C
EnsureNoModificationsHaveBeenDone();
EnsureSequentialModeMetadataOperations();
```